### PR TITLE
working react example

### PR DESCRIPTION
**Pull request by Vilhelm-Ian**

### CODE REVIEW

File: examples/mupdf-react/src/App.css [hunk 1]
```diff
@@ -1,38 +1,216 @@
...
+/* LAYOUT */
+
+html {
+  margin: 0;
+  padding: 0;
+  width: 100%;
+  height: 100%;
+}
+
+body {
```
Comment (jamie-lemon) [inline examples/mupdf-react/src/App.css:86]: Change this to `.App` to target the React DOM class for better layout

File: examples/mupdf-react/src/App.css [hunk 2]
```diff
@@ -1,38 +1,216 @@
...
+  padding: 0;
+  display: grid;
+  grid-template-columns: auto minmax(0, 1fr);
+  grid-template-rows: auto minmax(0, 1fr) auto;
+  width: 100%;
+  height: 100%;
+}
+
+header {
+  position: relative;
```
Comment (jamie-lemon) [inline examples/mupdf-react/src/App.css:97]: Change to:
```
postion: sticky
top:0
z-index:1
```
For better layout

File: examples/mupdf-react/src/App.css [hunk 3]
```diff
@@ -1,38 +1,216 @@
...
+  z-index: 500;
+}
+
+menu li {
+  padding: 4px 8px;
+  cursor: pointer;
+}
+
+/* OUTLINE */
+
```
Comment (jamie-lemon) [inline examples/mupdf-react/src/App.css:152]: Add a new DOM target with:
```
#outline-panel {
  position: fixed;
  top: 25px;
  height: 100vh;
  overflow-y: scroll;
}
```
For better layout :)

File: examples/mupdf-react/src/viewer.js [hunk 4]
```diff
@@ -0,0 +1,782 @@
...
+
+async function open_document_from_buffer(buffer, magic, title) {
+	current_doc = await worker.openDocumentFromBuffer(buffer, magic)
+
+	document.title = await worker.documentTitle(current_doc) || title
+
+	page_count = await worker.countPages(current_doc)
+
+	// Use second page as default page size (the cover page is often differently sized)
+	var page_size = await worker.getPageSize(current_doc, page_count > 1 ? 2 : 1)
```
Comment (jamie-lemon) [inline examples/mupdf-react/src/viewer.js:722]: Out of index by 1. Change to `1 ? 1 : 0` ( note this is also a bug in the current simple viewer, https://github.com/ArtifexSoftware/mupdf.js/commit/91228c47b452d27e9e106ff5353afb9e6fe26f63 )

File: examples/mupdf-react/src/App.js [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+        <details>
+          <summary>Edit</summary>
+          <menu>
+            <li onClick={() => Viewer.show_search_panel()}>Search...</li>
+          </menu>
+        </details>
+        <details>
+          <summary>View</summary>
+          <menu>
+            <li onClick={()=>Viewer.toggle_fullscreen()}>Fullscreen</li>
```
Comment (jamie-lemon) [inline examples/mupdf-react/src/App.js:37]: Weird indentation from here ....

**Review by jamie-lemon — CHANGES_REQUESTED** [review]
Added some layout tips which should help.

The main thing though is - can we somehow reference what would exist in `../../dist` for the libraries? We don't really want to duplicate this in `public`. If there is no way around it then update the README to inform the user about what they need to copy into "public".

**Comment by Vilhelm-Ian** [comment]
> Added some layout tips which should help.
> 
> The main thing though is - can we somehow reference what would exist in `../../dist` for the libraries? We don't really want to duplicate this in `public`. If there is no way around it then update the README to inform the user about what they need to copy into "public".

I couldn't find a way to get it to work without copying every file from dist

**Pull request closed without merging**